tools/cli/NearObjectCliHandler.cxx

```diff
@@ -17,6 +16,14 @@ void
 NearObjectCliHandler::SetParent(NearObjectCli* parent)
 {
     m_parent = parent;
+
+    auto controlFlowContext = (m_parent != nullptr) ? m_parent->GetControlFlowContext() : nullptr;
+    m_sessionEventCallbacks = std::make_shared<NearObjectCliUwbSessionEventCallbacks>([this, controlFlowContext = std::move(controlFlowContext)]() {
+        if (controlFlowContext != nullptr) {
+            controlFlowContext->OperationSignalComplete();
+        }
+        m_activeSession.reset();
+    });

```

Comment:
Even though the callbacks make use of data obtained from the parent, the callbacks should still be created if `SetParent()` is not called. So, I think the assignment to `m_sessionEventCallbacks` needs to go somewhere else. I don't see why it can't be done in the c'tor; the lambda passed to the `NearObjectCliUwbSessionEventCallbacks` c'tor will capture local vars like `parent` and will resolve their values at the time the lambda is invoked.

However, what's here is good enough for what we need right now, so let's revisit this later (so go ahead and resolve this comment without taking any action).